src/goto-symex/renaming_level.h

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Renaming levels
+
+#ifndef CPROVER_GOTO_SYMEX_RENAMING_LEVEL_H
+#define CPROVER_GOTO_SYMEX_RENAMING_LEVEL_H
+
+#include <map>
+#include <unordered_set>
+
+#include <util/irep.h>
+#include <util/ssa_expr.h>
+
+struct renaming_levelt
+{
+  virtual ~renaming_levelt() = default;
+
+  typedef std::map<irep_idt, std::pair<ssa_exprt, unsigned>> current_namest;
+  current_namest current_names;
+
+  unsigned current_count(const irep_idt &identifier) const
+  {
+    const auto it = current_names.find(identifier);
+    return it == current_names.end() ? 0 : it->second.second;
+  }
+
+  void increase_counter(const irep_idt &identifier)
+  {
+    PRECONDITION(current_names.find(identifier) != current_names.end());
+    ++current_names[identifier].second;
+  }
+
+  void get_variables(std::unordered_set<ssa_exprt, irep_hash> &vars) const
+  {
+    for(const auto &pair : current_names)
+      vars.insert(pair.second.first);
+  }
+};
+
+// level 0 -- threads!
+// renaming built for one particular interleaving
+struct symex_level0t : public renaming_levelt
+{
+  void
+  operator()(ssa_exprt &ssa_expr, const namespacet &ns, unsigned thread_nr);
+
+  symex_level0t() = default;
+  ~symex_level0t() override = default;
```

Comment:
There may subtleties I don't fully understand there, as to why there are needed. The constructor and destructor were defined before so I left the declaration, but then clion recommends to use `= default` instead of `{}` because it is more explicit (for the compiler in particular). Tagging @LAJW in case he can bring more insight here.

Reply:
It just looks a bit strange that we're overriding a method to do exactly what its parent method does, but if it was the same before then any decision to change it can be made in a future PR.